web/publicendpointresource.go

```diff
@@ -291,118 +291,35 @@ func restAddPort(w *rest.ResponseWriter, r *rest.Request, client *node.ControlCl
 	}
 
 	glog.V(0).Infof("Add PORT with %s %#v", r.URL.Path, r)
-	glog.V(0).Infof("request = %#v", request)
-
-	// Validate the port number
-	scrubbedPort := svc.ScrubPortString(request.PortName)
-	portParts := strings.Split(scrubbedPort, ":")
-	if len(portParts) <= 1 {
-		err := fmt.Errorf("Invalid port address. Port address be \":[PORT NUMBER]\" or \"[IP ADDRESS]:[PORT NUMBER]\"")
-		glog.Error(err)
-		restServerError(w, err)
-		return
-	}
-	port, err := strconv.Atoi(portParts[1])
```

Comment:
These were left over from an original mock up that Matos did.  He would probably know better why 5000 and 8080 were restricted.  I *think* this was an attempt to prevent them from adding an endpoint to a port that CC already uses for other things.  Since the code checks if the port is open already, that should take care of these (and others).. but more importantly a lot of those ports are only opened on 127.0.0.1, not all interfaces.  It should be valid to open port 5000 on a different interface.